src/test/java/com/github/jasminb/jsonapi/models/Article.java

```diff
@@ -12,7 +12,7 @@
 
 @Type("articles")
 @JsonIdentityInfo(generator = ObjectIdGenerators.StringIdGenerator.class, property = "id")
-public class Article {
+public class Article implements Commentable {
```

Comment:
I would prefer separate classes and test cases which cover both allowing/disallowing unknown types.